100+ Gigabit Ethernet in FPGA (#1)
jakubcabal/fpgahero.com: content/blog/20241126-100plus-gigabit-ethernet-in-fpga.md

---
title: "100+ Gigabit Ethernet in FPGA"
date: "2024-11-25"
author: "Jakub Cabal"
tags: 
  - "ethernet"
  - "hard-ip"
---

Ethernet is the technology that allows us to connect individual devices to networks and the entire Internet. 
Different devices use Ethernet for communication, remote control and also for data transfer.
It is therefore no coincidence that Ethernet support is also found in a wide range of FPGAs.

Implementing 100 Mbps Ethernet support is not difficult even on the cheapest FPGAs,
but higher speeds may require more sophisticated hardware.

## There is no Ethernet like Ethernet

Under the word Ethernet there are also many standards that differ not only in transmission speed.
Some standards are designed for metallic cables, others for fiber optics.

> For example, 100 Gigabit Ethernet can be transmitted over four serial lines or just two lines.

Reviewer: krátké vysvětlení (třeba jen takhle v závorce) toho, co jsou ty sériové linky by mohlo pomoct

> The difference may also be that some standards require RS-FEC (Reed–Solomon error correction, there are different variants) and others do not.

Reviewer: ```suggestion
The difference may also be that some standards require a certain variation of RS-FEC (Reed–Solomon Forward Error Correction) and others do not.
```

> There are a lot of these Ethernet standards,

Reviewer: There is a lot of
nebo
There are many

for an example you can look at the [list of just those for 100 Gbps transmission speeds](https://en.wikipedia.org/wiki/100_Gigabit_Ethernet#100G_interface_types).

> The engineer should know exactly what he will be working with, but it is not easy to know.

Reviewer: she/he
nebo dnes je populární
they

Reviewer: a ještě bych upravil ten konec, třeba: but sometimes, it's not easy to know ahead.

> Fortunately, there are only a few widely used Ethernet standards and we don't see many others in practice.

Reviewer: ```suggestion
Fortunately, only a few widely used Ethernet standards exist, and we don't see many others in practice.
```

## Multi Gigabit Ethernet support may vary

Ethernet support in FPGAs can vary not only in transmission speed,

> but also the level of built-in support (within the Hard IP blocks).

Reviewer 2: also in the level of

> but also the level of built-in support (within the Hard IP blocks).

Reviewer: Tu přišla zmínka Hard IP bloků úplně zčista jasna. Nebylo by lepší nejprve napsat, že některá FPGA tyto Hard IP celky obsahují. Nebo jen tuto větu přeformulovat, nicméně takhle je to trochu divné

> Some FPGAs have only fast enough serial lines (SERDES or also referred to as transceivers) to support proper encoding (NRZ, PAM4, etc.),

Reviewer 2: Some FPGAs have serial lines fast enough (SERDES or also referred to as transceivers) to only support proper encoding (NRZ, PAM4, etc.),

Reviewer: nelíbí se mi tu slovo "proper". Co má v tomto kontextu vyjadřovat?

> and therefore allow to implement PMA layer Ethernet support.

Reviewer: ```suggestion
and therefore allow the implementation of PMA support, which is the lowest sublayer of the Ethernet protocol.
```

Reviewer: možná bych tam vynechal i  slovo "support"

Better FPGAs have built-in logic to decode the Ethernet protocol at PCS level to the [Media-Independent Interface (MII)](https://en.wikipedia.org/wiki/Media-independent_interface) interface.

> However, the main trend is to provide full support, and so the best FPGAs have dedicated logic for all key Ethernet protocol layers including MAC and sometimes optional RS-FEC layer.

Reviewer: ```suggestion
However, the main trend is to provide full support, so the best FPGAs have dedicated logic for all key Ethernet protocol layers including MAC and sometimes the optional RS-FEC layer.
```

**So once again, briefly to summarize:**

- PMA - support only at the high-speed serial lines (SERDES) level

> - PMA+PCS - support including decoding to MII interface level

Reviewer: ```suggestion
- PMA+PCS - support further decoding to the MII interface
```

- PMA+PCS+MAC+(RS-FEC) - full Ethernet protocol support

> If your FPGA has support for at least the PAM layer, you can implement the remaining ones in the gate array.

Reviewer 2: PMA layer

Such solutions (in the form of Soft IP) are provided by FPGA chip manufacturers directly or by other companies for a considerable licensing fee.

## Compare 100+ Gigabit Ethernet Hard IPs

> Among FPGA chip manufacturers, AMD (Xilinx) and Altera (Intel) are the two main companies dominating.

Reviewer 2: To "dominating" na konci vety redundantné. Alebo by som prehodil slovosled na "two main dominating companies". Ale skôr by si vybral buď main alebo dominating.

> Among FPGA chip manufacturers, AMD (Xilinx) and Altera (Intel) are the two main companies dominating.

Reviewer: ```suggestion
Among FPGA chip manufacturers, AMD (previously Xilinx) and Altera (previously Intel) are currently the two main competitors.
```

Both manufacturers now have 100 Gigabit Ethernet-enabled FPGAs in their lineup and
occasionally we can also find FPGAs that can even handle 400 Gigabit Ethernet.

In the area of large FPGAs, Achronix is trying to get between the two manufacturers with its [Speedster7t](https://www.achronix.com/product/speedster7t-fpgas) chip,
which also has support for up to 400 Gigabit Ethernet.
However, its documentation is very weak so far and so I will focus on the two main manufacturers for now.

Let's conclude this post with a table comparing the available 100+ Gigabit Ethernet Hard IPs:

> | Vendor | Hard IP name |  FPGA family    | Supported speeds | RS-FEC | User interface |

Reviewer: Trochu se u méchají pojmy Vendor a Manufacturer. Nechám ale na tobě, jestli to chceš sjednotit
